frontend/src/components/module-editor/SideBar.tsx

```diff
@@ -1,46 +1,126 @@
-import React from "react";
+import React, { useContext } from "react";
+import { useParams } from "react-router-dom";
 import {
   Box,
+  Button,
   Flex,
+  Link,
+  HStack,
   Tabs,
   TabList,
   Tab,
   TabPanels,
   TabPanel,
+  Text,
 } from "@chakra-ui/react";
+import { ChevronLeftIcon, EditIcon } from "@chakra-ui/icons";
 
 import ModuleOverview from "./SideBarModuleOverview";
 import ContentKiosk from "./SideBarContentKiosk";
+import { ReactComponent as SaveIcon } from "../../assets/Save.svg";
+import { ADMIN_DASHBOARD_PAGE } from "../../constants/Routes";
+import EditorContext from "../../contexts/ModuleEditorContext";
+import { ModuleEditorParams } from "../../types/ModuleEditorTypes";
 
 const Sidebar = (): React.ReactElement => {
+  const { moduleIndex }: ModuleEditorParams = useParams();
+
+  const context = useContext(EditorContext);
+
+  if (!context) return <></>;
+  const { state } = context;
   return (
-    <Box w="20%">
+    <Box w="20%" minW="300px">
```

Comment:
not a big fan of setting the min width to a pixel value. Does it not work with `min-content` or `max-content`?